### PR TITLE
Ref #20: Cleanup warnings part 1

### PR DESCRIPTION
**Pull request by cndouglas**
## Pull Request Checklist

- [x] My patch has gone through review and I have addressed review comments
- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [x] I have updated the *Unit Tests* to cover new or changed functionality
- [x] I have updated the *UI Tests* to cover new or changed functionality
- [x] I have marked the bug with `[needsuplift]`
- [x] I have made sure that localizable strings use `NSLocalizableString()`

## Notes

Part 1 of #20

This cleans up a few dozen of the warnings that are easiest to fix. Summary of changes:

- Change `flatMap` (deprecated) to `compactMap`
- Change `!` to `?` in method signatures
- Add `String(describing:)` for string interpolation of optionals
- Convert `string.substring` (deprecated) to modern subscripts or `string.suffix(with:)`
- Remove unnecessary casting
- Remove `weak` from variables in protocol declarations
- Add `@discardableResult` where needed
- Change from `string.characters` (deprecated) to just `string`
- Remove unused `let error` in `catch`
- Change `UIApplication.shared.openURL()` (deprecated in iOS 10.0) to `UIApplication.shared.open()`
- Include `self.init()` in UIEdgeInsets initializer

### CODE REVIEW

File: Data/sync/Sync.swift [hunk 1]
```diff
@@ -449,7 +449,7 @@ extension Sync {
                     
                 // TODO: Needs favicon
                 if clientRecord == nil {
-                    recordType.coredataModelType?.add(rootObject: fetchedRoot, save: false, sendToSync: false, context: context)
+                    _ = recordType.coredataModelType?.add(rootObject: fetchedRoot, save: false, sendToSync: false, context: context)
```
Comment (jhreis) [inline Data/sync/Sync.swift]: We should add `@discardableResult` to the `add` function here, so we don't need to explicitly ignore it (e.g. `_ = `

File: Client/Frontend/Browser/TabManager.swift [hunk 2]
```diff
@@ -760,7 +760,7 @@ extension TabManager {
         }
         
         // Clean up any screenshots that are no longer associated with a tab.
-        imageStore?.clearExcluding(savedUUIDs)
+        _ = imageStore?.clearExcluding(savedUUIDs)
```
Comment (jhreis) [inline Client/Frontend/Browser/TabManager.swift]: Same, add `@discardableResult` to `clearExcluding`

**Review by jhreis — CHANGES_REQUESTED** [review]
Wow, this is great, thanks for hammering a lot of these out! Just one small thing (in two places). I know you didn't mark this as upliftable, yet, but figured I'd give it a quick run through anyway 😄

File: Client/Frontend/Browser/TabManager.swift (hunk 2 above)
Comment (cndouglas) [reply Client/Frontend/Browser/TabManager.swift]: Done.

File: Data/sync/Sync.swift (hunk 1 above)
Comment (cndouglas) [reply Data/sync/Sync.swift]: Done. Interestingly, the warning does not go away. Appears to be a bug with Xcode.

**Comment by cndouglas** [comment]
@jhreis When you say "mark this as upliftable", do you mean add the `needsuplift` label to the issue? I can't add labels. 😉

**Review by jhreis — APPROVED** [review]

**Comment by jhreis** [comment]
@Liunkae Totally forgot about that. We'll update the PR template on this, thanks for pointing that out 😄

**Comment by jhreis** [comment]
First community PR for Brave on this repo 😄Congrats!! 🎉

**Pull request merged**